forc-plugins/forc-client/src/op/deploy.rs

```diff
@@ -251,48 +264,138 @@ async fn deploy_new_proxy(
+    if scripts_to_deploy.is_empty() {
+        return Ok(deployed_scripts);
+    }
+
+    // Ensure that all packages are being deployed to the same node.
+    let node_url = get_node_url(
+        &command.node,
+        &scripts_to_deploy[0].descriptor.manifest_file.network,
+    )?;
+    if !scripts_to_deploy.iter().all(|pkg| {
+        get_node_url(&command.node, &pkg.descriptor.manifest_file.network).ok()
+            == Some(node_url.clone())
+    }) {
+        bail!("All scripts in a deployment should be deployed to the same node. Please ensure that the network specified in the Forc.toml files of all contracts is the same.");
+    }
+
+    let provider = Provider::connect(node_url).await?;
+
+    let wallet_mode = if command.default_signer || command.signing_key.is_some() {
+        SignerSelectionMode::Manual
+    } else if let Some(arn) = &command.aws_kms_signer {
+        SignerSelectionMode::AwsSigner(arn.clone())
+    } else {
+        println_action_green("", &format!("Wallet: {}", default_wallet_path().display()));
+        let password = prompt_forc_wallet_password()?;
+        SignerSelectionMode::ForcWallet(password)
+    };
+
+    // We will have 1 transaction per script as each script deployment uses a single blob.
+    let tx_count = scripts_to_deploy.len();
+    let account = select_account(
+        &wallet_mode,
+        command.default_signer || command.unsigned,
+        command.signing_key,
+        &provider,
+        tx_count,
+    )
+    .await?;
+
+    for pkg in &scripts_to_deploy {
+        let script = Executable::from_bytes(pkg.bytecode.bytes.clone());
+        let loader = script.to_loader();
+        println_action_green("Uploading", "blob containing script bytecode.");
+        loader.upload_blob(account.clone()).await;
+        println_action_green("Uploaded", "blob containing script bytecode.");
+
+        let loader_bytecode = loader.code();
+        let deployed_script = DeployedScript {
+            bytecode: loader_bytecode,
+        };
+        deployed_scripts.push(deployed_script);
+    }
+    Ok(deployed_scripts)
+}
+
+/// Builds and deploys contract(s). If the given path corresponds to a workspace, all deployable members
+/// will be built and deployed.
+///
+/// Upon success, returns the ID of each deployed contract in order of deployment.
+///
+/// When deploying a single contract, only that contract's ID is returned.
+pub async fn deploy_contracts(
+    command: &cmd::Deploy,
+    contracts_to_deploy: Vec<Arc<BuiltPackage>>,
+) -> Result<Vec<DeployedContract>> {
+    let mut deployed_contracts = Vec::new();
+
+    if contracts_to_deploy.is_empty() {
         return Ok(deployed_contracts);
     }
 
     let contract_salt_map = if let Some(salt_input) = &command.salt {
         // If we're building 1 package, we just parse the salt as a string, ie. 0x00...
         // If we're building >1 package, we must parse the salt as a pair of strings, ie. contract_name:0x00...
```

Comment:
We may be able to replace this `if/else` with just a single block. I think the reason behind the split is the `split_once()` inside `validate_and_parse_salts()`... but if we change `validate_and_parse_salts()` to use `split().next()` then I'm hoping these two sides of the conditional will be the same.

Here's a test: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021

Reply:
Hey @alfiedotwtf i'll take a look closely but the link seems to send me to default/empty playground

Reply:
Lol sorry, I forgot to permalink it - https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=6f044ab71e624d696967326527f1a0d9

Reply:
Hey @alfiedotwtf, sorry I was just able to return to this one. I think given that this section not introduced with this PR and also the criticality of this specific PR, I am inclined not to touch it in this PR but add this to #6259, if you are also fine with it